### DISCUSSION TITLE
error while loading shared libraries: libglog.so.1: cannot open shared object file: No such file or directory

### DISCUSSION THREAD

**Discussion by luger1990** [apache/kvrocks#615] [Q&A]
i want to build kvrocks once, and scp  './build/kvrocks' file to other machines. but at other machines if i start kvrocks, it shows 'error while loading shared libraries: libglog.so.1: cannot open shared object file: No such file or directory'
should i build at every machine? and it it too large size,like below:
```
8.0K	CMakeLists.txt
8.0K	CODE_OF_CONDUCT.md
4.0K	CONTRIBUTING
12K	Changelog
4.0K	DISCLAIMER
4.0K	Dockerfile
12K	LICENSE
4.0K	NOTICE
8.0K	README.md
2.6G	build
4.0K	build.sh
56K	cmake
4.0K	cppcheck.sh
4.0K	cpplint.sh
320K	docs
28K	kvrocks.conf
8.0K	licenses
4.0K	package.sh
1.1M	src
8.0K	systemd
908K	tests
124K	tools
```
the fold of build is 2.6G, it is too large.

```
root@bhh77n3:/tmp/incubator-kvrocks/build/_deps# find . -name libglog.so.1
./glog-build/libglog.so.1
```

**Comment by PragmaTwice** [comment]
Hello~ Could you provide the commit hash / tag of this repo that you use to build kvrocks?

**Reply by PragmaTwice** [reply]
AFAIK we use static library of glog in the head commit b9f7863abc3d15f5a489c7c210eb83712c3cf05e of `unstable`, so this problem may not happen.

**Reply by PragmaTwice** [reply]
If you use other commit where makefiles still exists, you can try to install glog using package manager in your system (e.g. apt, pacman, yum), which may bring a `libglog.so` to your system lib path.

**Reply by luger1990** [reply]
thanks i will try this method

**Reply by PragmaTwice** [reply]
Sorry, shared library of glog is used in current head commit, see @git-hulk's comment to check the details 😢

**Comment by git-hulk** [comment]
@PragmaTwice I tried the latest unstable branch on Ubuntu, glog is linking as the dynamic library:
```
ty.lin@build-clickhouse:~/incubator-kvrocks/build$ ldd kvrocks
	linux-vdso.so.1 (0x00007ffcbfb5d000)
	libglog.so.1 => /home/ty.lin/incubator-kvrocks/build/_deps/glog-build/libglog.so.1 (0x00007fea007ff000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fea007b5000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fea00666000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fea00474000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fea0143b000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fea0025a000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fea0023d000)
```

Need to diable the share library explicitly in `glog.cmake`.

**Reply by PragmaTwice** [reply]
Oh sorry, I thought it was a `BUILD_SHARED_LIBS=OFF` in glog.cmake, but I find that it is in snappy.cmake not in glog.cmake.

**Reply by PragmaTwice** [reply]
A `BUILD_SHARED_LIBS=OFF` in glog.cmake is enough, the one in snappy.cmake is not necessary.

**Reply by git-hulk** [reply]
@PragmaTwice Would you like to fix this issue?

**Reply by PragmaTwice** [reply]
I am glad to do that, maybe at a later date in this week : )

**Reply by Geo-fortune** [reply]
You can resolve this issue by compiling all artifacts as dylib.

**Reply by git-hulk** [reply]
@Geo-fortune It’s a bug that didn’t static link the glow and fixed in latest release.
